test/Microsoft.ML.Predictor.Tests/TestAutoInference.cs

```diff
@@ -272,6 +272,259 @@ public void EntryPointPipelineSweep()
             Assert.True(rows.All(r => r.TrainingMetricValue > 0.1));
         }
 
+        [Fact]
+        [TestCategory("EntryPoints")]
+        public void EntryPointPipelineSweepColumnPurposeDefaults()
+        {
+            // Get datasets
+            var pathData = GetDataPath("adult.train");
+            var pathDataTest = GetDataPath("adult.test");
+            const int numOfSampleRows = 1000;
+            int numIterations = 4;
+            const string schema =
+                "sep=, col=age:R4:0 col=fnlwgt:R4:2 col=education_num:R4:4 col=Features:R4:10-12 col=workclass:TX:1 col=education:TX:3 col=marital_status:TX:5 col=occupation:TX:6 " +
+                "col=relationship:TX:7 col=ethnicity:TX:8 col=sex:TX:9 col=native_country:TX:13 col=IsOver50K:R4:14 header=+";
+            var inputFileTrain = new SimpleFileHandle(Env, pathData, false, false);
+#pragma warning disable 0618
+            var datasetTrain = ImportTextData.ImportText(Env,
+                new ImportTextData.Input { InputFile = inputFileTrain, CustomSchema = schema }).Data.Take(numOfSampleRows);
+            var inputFileTest = new SimpleFileHandle(Env, pathDataTest, false, false);
+            var datasetTest = ImportTextData.ImportText(Env,
+                new ImportTextData.Input { InputFile = inputFileTest, CustomSchema = schema }).Data.Take(numOfSampleRows);
+#pragma warning restore 0618
+            // Define entrypoint graph
+            string inputGraph = @"
```

Comment:
I know what most of tests written in way of "Create manually json string and execute it", but it doesn't mean it has to be this way.
check TestSimpleExperiment 
any chance I can convince you to use C# classes instead of composing JSON string manually?

Reply:
I added a 4th test case which follows this paradigm.  Does it look good ?  

The other 3 tests I would like to keep as is if possible

---
In reply to: [199594792](https://github.com/dotnet/machinelearning/pull/461#discussion_r199594792) [](ancestors = 199594792)